controllers/webhook/webhook.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```

Comment:
Not really about this PR per se, but we are inconsitent with adding LICENSE headers across the project. Some of the files have it whereas other do not. 

My personal opinion is that having LICENSE file in the root of the project is sufficient.

The benefits I see:
- less redundant lines 
- no need to ensure consistency across all the files (ideally through automation)
- less scrolling when reviewing PRs on GitHub :)

I created an issue to track it.